research/integrations/fedora-ci/index.md

```diff
@@ -0,0 +1,338 @@
+      "kickstart": null
+    }
+  ],
+  "notes": [
+    {
+      "level": "info",
+      "message": "tf-tmt/dispatch-1725615078-80d05f05"
+    }
+  ],
+  "result": {
+    "summary": null,
+    "overall": "passed",
+    "xunit": "…",
+    "xunit_url": "https://artifacts.dev.testing-farm.io/2fdee1a0-0746-45ad-ad0c-d3ec8833630b/results.xml"
+  },
+  "run": {
+    "console": null,
+    "stages": null,
+    "artifacts": "https://artifacts.dev.testing-farm.io/2fdee1a0-0746-45ad-ad0c-d3ec8833630b"
+  },
+  "settings": null,
+  "user": {
+    "webpage": null
+  },
+  "queued_time": 6.892783,
+  "run_time": 1207.608357,
+  "created": "2024-09-06T09:31:17.471815",
+  "updated": "2024-09-06T09:31:17.471844"
+}
+```
+
+Most notable differences:
+
+- RPMs are provided from Koji scratch build (following Fedora CI, which is
+  easier than self-hosting)
+- chroot is defined by the branch rather than config
+
+## Technical issues / questions
+
+### Load balancing
+
+Preferably it would be ideal to have separate deployment for the “dist-git only”
+service as handling all Fedora packages would definitely increase traffic and
+could cause issues with running as both Fedora CI and original Packit Service.
+
+#### Default config
+
+Having a separate instance would also open up a door to a potential “default
+config” which could be different to our current one (SRPM, Copr build, TF) and
+more tailored to the needs of the Fedora CI.
+
+**TODO**:
+
+- [ ] Maybe have a default config configurable?
+      For context, we have [default jobs](https://packit.dev/docs/guide#jobs) on
+      the upstream in case there are no jobs specified. This could be leveraged
+      to provide some **reasonable** defaults for Koji builds and Testing Farm
+      runs on the downstream part.
+
+### SRPM build
+
+For Zuul is run via `fedpkg srpm`. Then it's provided for the Koji build.
+
+Packit should not interefere with the way the SRPM gets built.
+
+### Propagating Koji build
+
+As of now, the sources need to be already uploaded in the lookaside cache to run
+the RPM build in general (either Fedora CI or Zuul CI). Given that merges in
+dist-git are handled as fast-forward, it should be possible to just tag the
+build from the PR instead of rebuilding again (with the same sources and
+specfile).
+
+:::note
+
+This is definitely a _nice-to-have_, a follow-up.
+
+:::
+
+**TODO**:
+
+- [ ] Does it make sense to open a follow-up already, even if there's no PoC
+      yet?
+
+### RPM for Testing Farm
+
+Zuul provides its own (self-hosted) repository. Fedora CI just refers to the
+scratch build.
+
+Self-hosting the repository is not preferred, following the Fedora CI approach
+might be the best.
+
+### STI tests
+
+Older format of tests that can be run via Zuul. Based on the TF API, Testing
+Farm supports it too. Requesting STI tests from the Testing Farm should be
+possible, detection of their presence lies on Packit.
+
+**TODO**:
+
+- [ ] Should we support the STI tests?
+
```

Comment:
We can ask TF if we shouldn't motivate people to go away by not supporting it...;)

Reply:
Is the presence detection of STI hard?

Reply:
> Is the presence detection of STI hard?

Have no clue :) I have **never** even seen the STI tests in the dist-git :D